components/dropdowns/profile/ProfileDropdown.tsx

```diff
@@ -0,0 +1,87 @@
+import { CalendarIcon, LogoutIcon, ProfileIcon, SettingsIcon, SupportIcon } from "@components/icons";
+import { useNavbarContext } from "@components/navbar/NavbarProvider";
+import { useCloseDropdown } from "@hooks/useCloseDropdown";
+import { FC, SVGProps, useRef, useState } from "react";
+import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react";
+import { useUserCartContext } from "@components/user/UserCartProvider";
+
+const ProfileDropdown = () => {
+  const { visible } = useNavbarContext();
+  const [showDropdown, setShowDropdown] = useState(false);
+
+  const user = useUser();
+  const firstName = user?.user_metadata.full_name.split(" ")[0];
+
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  useCloseDropdown<HTMLDivElement>(wrapperRef, setShowDropdown);
+
+  return (
+    <div ref={wrapperRef} className="relative">
+      <button className="inline-flex items-center gap-3" onClick={() => setShowDropdown((oldValue) => !oldValue)}>
+        <div className="max-w-[140px] select-none overflow-hidden text-ellipsis whitespace-nowrap text-dark">
+          {firstName}
+        </div>
+        <div className="h-10 w-10 rounded-full bg-gradient-to-r from-indigo-300 to-purple-400" />
+      </button>
+      <ProfileDropdown.Content visible={showDropdown && visible} />
+    </div>
+  );
+};
+
+export default ProfileDropdown;
+
+ProfileDropdown.Content = function ({ visible }: { visible: boolean }) {
```

Comment:
I see why you did the `PrimaryComponent.ChildComponent` patterns, but it feels unnecessary here.

You are exposing these back to the user to use but you only use these internally and the Profile Dropdown does not accept children.

But it's a good implementation